data/caasp/update.sh

```diff
@@ -123,12 +143,56 @@ elif [ ! -z "${INSTALL:-}" ]; then
         fi
     done < qam_packages.txt
 
+    if [ ${reboot_required} -eq 1 ]; then
+        # Disable the update repo, so to check if this package is old or new
+        $runner "zypper mr -d UPDATE"
```

Comment:
That is correct. I am using admin as a reference point for getting the status of the packages (*present* or *not*), and based on that I apply changes for the whole cluster. e.g.

```bash
echo "The following new packages are going to be installed: $(cat new_package.txt)"
$runner "/usr/sbin/transactional-update salt pkg install -y $(cat new_package.txt)"
```

In other words, I am assuming if `pkg_a` and `pkg_b` are not installed in the admin node, it would most probably not be installed also for the other nodes. So far, the risk of this *probability* works fine. This might not be 100% correct, but it's better than not installing the packages at all. I agree with you that there's a better a approach, which I would like to work on later on (refactoring).